### ISSUE TITLE
exec "$foo": prints fish internal unicode private use character U+FDD3

### ISSUE THREAD

**Issue by floam** [fish-shell/fish-shell#3187]
### Reproduction Steps:

Where `test` can be any name that is not a set variable:
1. Type `> exec "$test"` ( or just `> "$test"`) 
2. (or just type a double quoted dollar sign and hit enter: `> "$"`)
3. Hit enter and observe the error message.
### Observed behavior:

![image](https://cloud.githubusercontent.com/assets/291142/16526190/d0be8f00-3f64-11e6-96e1-820278ab9558.png)
Also, same thing happens to `$argv` on the command not found handler. Github isn't rendering the character below but it is between the quotes:

```
> "$"
fish: Unknown command ''
```
### Additional information:

Doesn't happen if one uses single quotes. Bug is in `master`, 2.3.0, 2.2.0.

**Comment by krader1961** [comment]
FYI, \uFDD3 is `VARIABLE_EXPAND_SINGLE`. The problem is in *parse_execution.cpp* where it tests if the first character of the command is the internal token that indicates the var name is quoted then proceeds to print that internal token.

**Comment by floam** [comment]
Looks like the same thing still happens, except for builtins: `fish -c "builtin \"\$foo\""`
```
<E> fish: Unknown builtin '﷓foo'
```

**Comment by krader1961** [comment]
Fine, @floam, but please open a new issue that references this one. The original problem was solved and shipped with the 2.4.0 release. You've find a new way to cause this problem.
